### PR TITLE
Finish: Using Redux in React components

### PR DESCRIPTION
**Pull request by Zafron047**
Hello Dear Reviewer,

This PR implements React-Redux integration in the `<App>` component, sets up the Redux store, and adds book management components, allowing users to display, add, and remove books. The changes are made as per the following requirements:

- [x] **React-Redux Integration:**
The `<App>` component is now wrapped with the `<Provider>` component from react-redux, ensuring that the Redux store is accessible throughout the application.

- [x] **Initial State for booksSlice:**
The initialState variable for the booksSlice has been created, containing an array of objects with the following properties:
```javascript
  const initialState = [
    {
      item_id: 'Item1',
      title: 'Book Title',
      author: 'Author',
      category: 'Category',
    },
  ],
``` 
- [x] **Reusable Component to Display Books:**
A reusable component named `<BookList>` has been created to display books. This component is utilized in the `<Library> `component to render the list of books received from the booksSlice.

- [x] **Dispatch Actions using useDispatch:**
The useDispatch hook is used to dispatch actions in the `<BookList>` component. It enables the functionality to remove a book from the booksSlice.

- [x] **Add Book Button Component:**
A `<Button>` component has been added, featuring an event handler to add a book to the `books `array, including attributes like `id`, `title`, and `author`.

- [x] **Remove Book Button Component:**
Another `<Button> `component has been implemented with an event handler to remove a book from the `books `array by its `id`.

These changes enable seamless management of books in the application using React-Redux integration and the provided components. All tests have been conducted to ensure the smooth functioning of the features.

Please review this pull request and provide feedback. Thank you!

Regards,
@Zafron047

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -1,23 +1,43 @@
...
-        author: action.payload.author,
-      });
+      const { title, author, category } = action.payload;
+      const newBook = {
+        item_id: `Item${state.length + 1}`,
+        title,
+        author,
+        category,
+      };
+      return [...state, newBook];
```
Comment (iambenkis) [inline src/redux/books/booksSlice.js:36]: - [ ] Keep doing the great work with the add and remove actions 👏 👏 👏

**Review by iambenkis — CHANGES_REQUESTED** [review]
#### Hi @Zafron047 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

giphy


## Highlights

✔️ Reducer and actions are okay
✔️ Provider is well included
✔️ Professional PR description

## Suggested changes

- [x] Kindly fix the broken link attached to the MIT license ,  it will make your documentation more professional 👍 

Capture

_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (Zafron047) [reply src/redux/books/booksSlice.js:36]: @iambenkis 

- [x] I have now added an MIT.md License to this branch of the project.
- [x] Gee 😄 Thanks for the appreciation I made the changes on the `add and remove actions` while paying another attention.

I wish you have a great day.

**Review by nedjwab — APPROVED** [review]
# 🟢 Project Approved! 🏆

## Hi  @Zafron047           👋

Your project is complete! ✔️✔  🏁

Your code was very professional 💯

There is nothing else to say other than... _it's time to merge it_ :shipit:
Congratulations! 🎉

 
## Highlights 🌟

- ✔️ Descriptive commmit messsages
- ✔️ Descriptive PR description
- ✔️ No linter Errors


- The code is top-notch! 💯 and have a huge potential.

Cheers and Happy coding!👏👏👏

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.


_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Zafron047** [comment]
@nedjwab Thanks a lot for the code review.

**Pull request merged**